react-common/components/controls/Feedback/FeedbackEventListener.ts

```diff
@@ -0,0 +1,125 @@
+import { appId, feedbackFrameUrl } from './configs';
+interface FeedbackRequestEventPayload<T> {
+  Event: string
+  EventArgs: string
+}
+type FeedbackRequestPayloadType = FeedbackRequestEventPayload<any>
+
+interface FeedbackResponseEventPayload<T> {
+  event: string
+  data: T
+  error?: any
+}
+
+// for styling the feedback, we use this object. It is mostly used to change the colors.
+// we'll want to change this based on the target and whether high contrast is enabled
+let themeOptions = {
+    baseTheme: "PublisherLightTheme",
+}
+
+let initfeedbackOptions: any;
+let feedbackData: any;
+let FEEDBACK_FRAME_ID: string;
+let currentTheme = '';
+let feedbackCallbacks: any;
+
+// the function to initialize the feedback event listener
+// feedbackConfig: needs to be passed in as a prop because the things that
+/**
+ * The function to initialize the feedback event listener
+ * @param {any} feedbackConfig: the feedback config object whose fields are defined in OCV.
+ *  This changes based on what type of feedback we want to collect. Look at configs.ts for more details.
+ * @param {string} frameId: the html id of the actual iframe where the feedback will be displayed
+ * @param {any} [callbacks]: an object of functions that can be called when certain events happen in the feedback modal.
+ *  Needs to be passed in because the callbacks will depend on what the parent wants to react to.
+ */
+export const initFeedbackEventListener = (feedbackConfig: any, frameId: string, callbacks?: any) => {
+    window.addEventListener('message', feedbackCallbackEventListener);
+    feedbackCallbacks = callbacks;
+    initfeedbackOptions = {
+        appId: appId,
+        ageGroup: "Undefined",
+        authenticationType: "Unauthenticated",
+        clientName: "MakeCode",
+        feedbackConfig: feedbackConfig,
+        isProduction: false,
+        themeOptions: themeOptions,
+        // telemetry - will likely want this
+    }
+
+    feedbackData = initfeedbackOptions;
+    FEEDBACK_FRAME_ID = frameId;
+}
+
+export const removeFeedbackEventListener = () => {
+    window.removeEventListener('message', feedbackCallbackEventListener);
+}
+
+/**
+ * The function that listens for the feedback events.
+ * The events here are the ones that seemed most useful to log or respond to
+ * @param {MessageEvent<FeedbackRequestPayloadType>} event: the event received from OCV
+ */
+const feedbackCallbackEventListener = (event: MessageEvent<FeedbackRequestPayloadType>) => {
```

Comment:
whitespace in this function seems messed up. the body isn't indented and neither is the switch statement